include/seastar/net/tls.hh

```diff
@@ -220,6 +220,24 @@ namespace tls {
          */
         void set_dn_verification_callback(dn_callback);
 
+        /**
+         * Retrieve information about loaded certificates.
+         *
+         * Returns a std::vector of cert_info, each extracted from a loaded
+         * certificate, std::nullopt if the impl does not exist or we detect
+         * an error during extraction.
```

Comment:
Shouldn't we throw on error? It's dangerous to just swallow errors, esp. in security. The caller can always ignore the exception if they want.

Reply:
Fair point. I was thinking that an exception might be overkill since these are effectively just accessors on a fully constructed credentials struct, but I suppose more information is better.